### ISSUE TITLE
Improve experience when CloudMap Namespace doesn't exist

### ISSUE THREAD

**Issue by nckturner** [aws/aws-app-mesh-controller-for-k8s#109]
Currently, the CloudMap Namespace has to be created out of band from the rest of the resources.  There should be an automated way to create it, whether its a separate custom resource like the Mesh (which can be auto created by the injector) or something else.

**Comment by kiranmeduri** [comment]
I am against this giving namespace can be of scope larger than a single Kubernetes cluster. One can argue the same for mesh CRD but we already added it. Specifically this can be a problem when there are services in same namespace that are deployed in different clusters or across compute platforms. Open to discuss this further.

**Comment by nckturner** [comment]
Yes, I think we should either work towards deprecating the mesh resource and expect users to create them out of band, or allow a single cluster to own the mesh and cloudmap namespaces.

**Comment by shubharao** [comment]
Investigate: should mesh creation take an input on "also create a namespace" 
Investigate: should app mesh entities be auto-created to map to existing K8s resources - cluster name, namespace name, services etc.

**Comment by astrived** [comment]
@achevuru is going to work on this issue.

**Comment by astrived** [comment]
This is working as designed, we will revisit this ask based on customer feedback.

**Comment by RafalMaleska** [comment]
are You planning to provide CloudMap as CRD?
